### PR TITLE
Added batch_workflows endpoint, unit tests, converted style to pep8 s…

### PR DESCRIPTION
**Pull request by Brett55**
Refactored unit tests, pep8 styling, and added batch workflow endpoints

### CODE REVIEW

**Comment by nricklin** [comment]
Looking good.  Can you please add:
- Example in the examples directory
- Documentation in the rich-text documentation files (that get auto-built in readthedocs)

**Comment by Brett55** [comment]
added example and documentation

**Comment by kstamatiou** [comment]
Hi Brett. Thanks a lot for contributing.

I think at this point we should remove the example in the docs all together.
gbdxtools is about abstracting away the hassle of dealing with json's. 

The machinery that you've created is necessary for the next step which is to create 
batch task objects which are strung together to create a batch workflow. In fact, instantiating the  Task class with array inputs instead of scalar inputs should automatically create a batch task object. Then, instantiating the Workflow class with batch task objects  should automatically create a batch workflow object. When we do workflow.execute(), the batch workflow endpoint is automatically hit.

I'll create a new issue on this.

In summary, remove the example and make another request (sorry you had to write the example for nothing!)

**Comment by Brett55** [comment]
I agree 100%, I think batch_workflows will be better utilized in the way
simple workflows is used. I will refactor and re-open the PR when done.

On Tue, May 10, 2016 at 12:06 PM, Kostas Stamatiou <notifications@github.com

**Comment by kstamatiou** [comment]
@Brett55 

If you don't want to do this, I can accept your pull request as is (with example removed) 
and either @nricklin or I will take a stab at issue #34.

If you want to do it, go for it!

**Comment by Brett55** [comment]
I'm more than happy to do it, if a user submits an array for a task input,
it will be easy to then hit the batch workflows endpoint.

**Comment by kstamatiou** [comment]
@Brett55 

Awesome!

**Comment by nricklin** [comment]
Thanks Brett!

We probably need logic something like this:
- if any input is an array, validate that each other input is EITHER an array of the same length, OR a singleton.
